PyMca5/PyMcaGui/io/TiledCatalogSelector.py

```diff
@@ -64,28 +64,28 @@ def __init__(
         self.url_validation_error = self.signals.url_validation_error
 
         # A buffer to receive updates while the URL is being edited
-        self.url_buffer = ""
+        self._url_buffer = self.url
 
-    def on_url_focus_in_event(self, event: QEvent):
-        """Handle the event when the URL widget gains focus."""
-        _logger.debug("TiledCatalogSelector.on_url_focus_in_event()...")
-
-        # TODO: Check whether this causes issues under the condition when
-        #       the connection button is clicked before the user
-        #       interacts with the URL text editor.
-        self.url_buffer = ""
+    @property
+    def url(self):
+        return self._url
+    
+    @url.setter
+    def url(self, value: str):
+        self._url = value
+        self._url_buffer = value
 
     def on_url_text_edited(self, new_text: str):
         """Handle a notification that the URL is being edited."""
         _logger.debug("TiledCatalogSelector.on_url_text_edited()...")
 
-        self.url_buffer = new_text
+        self._url_buffer = new_text
 
     def on_url_editing_finished(self):
         """Handle a notification that URL editing is complete."""
         _logger.debug("TiledCatalogSelector.on_url_editing_finished()...")
 
-        new_url = str.strip(self.url_buffer or "")
+        new_url = str.strip(self._url_buffer or "")
```

Comment:
Now that `self._url_buffer` will always be a string (and not `None`), we should be able to go back to this simpler version.

```suggestion
        new_url = self._url_buffer.strip()
```